perf/benchmark/security/generate_policies/README.md

```diff
@@ -29,21 +33,100 @@ spec:
        - 0.0.0.0
 ```
 
-generate_polices allows to customize the generated policies with command line flags:
+The flags which can be used to create custom AuthorizationPolicies are as follows:
+
+```bash
+  AuthorizationPolicies:int
+  namespace:string
+  action:DENY/ALLOW
+  numPolicies:int
+  numPaths:int
+  numValues:int
+  numSourceIP:int
+  numNamespaces:int
+  numPrincipals:int
+```
+
+For more information see [AuthorizationPolicy Reference](https://istio.io/latest/docs/reference/config/security/authorization-policy/).
+
+## PeerAuthentication
+
+To create a PeerAuthentication policy one must explicity pass in the following flag -generate_policy="PeerAuthentication:1".
+
+To create a PeerAuthentication, run the following command:
+
+```bash
+go run generate_policies.go generate.go -generate_policy="PeerAuthentication:1"
+```
+
+This will create a PeerAuthentication policy as follows since mtlsMode has a default value of STRICT and print it out to the stdout. The number after "PeerAuthentication" is used to determine how many PeerAuthentication will be produced to stdout.
+
+```yaml
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: test-PeerAuthentication-1
+  namespace: twopods-istio
+spec:
+ mtls:
+   mode: STRICT
+```
+
+The flags which can be used to create custom PeerAuthentication are as follows:
 
 ```bash
-Optional arguments:
-  -h, --help
-  -generate_policy string         List of key value pairs separated by commas.
-                                  Supported options: namespace:string, action:DENY/ALLOW, policyType:AuthorizationPolicy, numPolicies:int, numPaths:int, numValues:int, numSourceIP:int, numNamespaces:int (default "numPolicies:1")
+  PeerAuthentication:int
+  namespace:string
+  mtlsMode:STRICT/DISABLE
 ```
 
-To create a large policy to an output .yaml file, run the following command:
+For more information see [PeerAuthentication Reference](https://istio.io/latest/docs/reference/config/security/peer_authentication/).
+
+## Examples
+
+generate_policies.go also allows a user to create mutliple kinds of policies in one command.
+To generate 1 AuthorizationPolicy with a principals rule and 1 PeerAuthorization policy with STRICT mtlsMode, run the following command:
 
 ```bash
-go run generate_policies.go generate.go -generate_policy="numSourceIP:1000,numPaths:1000,numNamespaces:1000" > largePolicy.yaml
+ go run generate_policies.go generate.go -generate_policy="AuthorizationPolicy:1,numPrincipals:1,PeerAuthentication:1"
```

Comment:
Can we use a well-defined YAML (or JSON) for the format of the `generate_policy`?

You will need to define a go struct with JSON annotations, and pass the YAML (or JSON) from the command line (or from a file) and the parse it to the go struct in the code?

Reply:
or just a key-value config file (like generate_policy.config) if those parameters in not nested

Reply:
I do think it would be cleaner to have some structure or config file. We can discuss this in our sync up

Reply:
Updated the code to allow for a go struct with JSON annotations, the flag now takes in the config file name. The README.md explains in more detail how this works